src/main/java/org/zalando/nakadi/domain/BatchItem.java

```diff
@@ -58,4 +59,8 @@ public void updateStatusAndDetail(final EventPublishingStatus publishingStatus,
         response.setPublishingStatus(publishingStatus);
         response.setDetail(detail);
     }
+
+    public int getEventSize() {
+        return event.toString().getBytes(StandardCharsets.UTF_8).length;
```

Comment:
@lmontrieux For me this approach is not so good.
At first we should decide which data we are storing to event repository.
If we are storing object(JSON), than it makes no sense to check it on validation step, because size of object means nothing on this step.
Probably we should pass items with strings or byte array to TopicRepository, at least these two types have estimation on size, TopicRepository should store this representation (string or byte[]), and this representation should be used on layer below (KafkaTopicRepositry)
So for me it's better to first validate, then partition, then enrich, than prepareToPublish. On step `prepareToPublish` one should convert nakadi representation to storage representation and validate size. In this approach we will avoid several toString converters, and we will be sure which kind of data is stored in TopicRepository.